pretraining/utils/dataset_utils.py

```diff
@@ -841,7 +852,12 @@ class Sampling_Dataset(_Stateful_Dataset):
     def __init__(
         self,
         datapath: str,
-        dataset: Type[_Stateful_Dataset],
+        dataset: Union[
+            Type[Streaming_Doc_Dataset],
+            Type[Shard_Doc_Dataset],
+            Type[Shard_Line_Dataset],
```

Comment:
FYI Shard_Doc and Shard_Line Datasets are about to be removed anyway:

https://github.com/foundation-model-stack/pretrain/pull/5/files#diff-060f427d7e3e19de03067dd05b75e3758ba40f69745b2afa58a6ffe63024f81eL1214

Reply:
Sure, they can be removed from here once they are actually removed.

Reply:
I removed them now.